### ISSUE TITLE
in Coyote lib., CGWINDOW is not exporting PDF + Raster as expected

### ISSUE THREAD

**Issue by alaingdl** [gnudatalanguage/gdl#1380]
How to test ? 
1/ Compile GDL with the script (and with wxwidgets, mandatory)
2/ add coyote lib in the PATH (git clone  https://salsa.debian.org/debian-astro-team/coyote)
https://github.com/gnudatalanguage/gdl/wiki/Third-party-librairies
3/ run GDL

```
GDL> CGWINDOW
GDL> % Compiled module: FILEPATH.
% Compiled module: CGPICKFILE.
% Compiled module: CGSOURCEDIR.
../src/common/filename.cpp(567): assert "volDummy.empty() && pathDummy.empty()" failed in Assign(): the file name shouldn't contain the path

...
```

Except a file  : comands.cgs, no Raster files written. PS fil was null.

![bug_cgwindow221018](https://user-images.githubusercontent.com/38264661/196479300-3bcbaa22-05ce-4673-8969-db94ca667c80.png)

**Comment by alaingdl** [comment]
ASSERT INFO:
../src/common/filename.cpp(568): assert "volDummy.empty() && pathDummy.empty()" failed in Assign(): the file name shouldn't contain the path

BACKTRACE:
[1] wxFileName::Assign(wxString const&, wxString const&, wxPathFormat)
[2] wxFileDialog::Create(wxWindow*, wxString const&, wxString const&, wxString const&, wxString const&, long, wxPoint const&, wxSize const&, wxString const&)
[3] wxFileDialog::wxFileDialog(wxWindow*, wxString const&, wxString const&, wxString const&, wxString const&, long, wxPoint const&, wxSize const&, wxString const&)
[4] lib::dialog_pickfile_wxwidgets(EnvT*)
[5] FCALL_LIB_RETNEWNode::Eval()
[6] ASSIGN_REPLACENode::Run()
[7] GDLInterpreter::statement(ProgNode*)
[8] GDLInterpreter::call_fun(ProgNode*)
[9] FCALLNode::Eval()
[10] ASSIGN_REPLACENode::Run()
[11] GDLInterpreter::statement(ProgNode*)
[12] GDLInterpreter::call_pro(ProgNode*)
[13] MPCALLNode::Run()
[14] GDLInterpreter::statement(ProgNode*)
[15] GDLInterpreter::call_pro(ProgNode*)
[16] CallEventPro(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, BaseGDL*, BaseGDL*)
[17] CallEventHandler(DStructGDL*)
[18] GDLWidget::HandleWidgetEvents()
[19] GDLEventHandler()
[20] rl_read_key
[21] readline_internal_char
[22] readline
[23] DInterpreter::GetLine[abi:cxx11]()
[24] DInterpreter::ExecuteLine(std::istream*, unsigned long long)
[25] DInterpreter::InterpreterLoop(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
[26] main
[27] __libc_start_main
[28] _start

**Comment by alaingdl** [comment]
This backtrace comes from a run on Debian 11. If I click "continue", I succeed to save a PNG and the PNG is OK !
(PNG writing not working on my U22.04)

**Comment by GillesDuvert** [comment]
Actually there are two problems, one in FSC_FILESELECT (?) and an assert in wxWidgets, the one you report.
Perhaps related to the order where actions are made: there should be a menu opening asking for the file name and sizes, that is completely messed up.

**Comment by GillesDuvert** [comment]
solved by #1384 at the current GDL level.
Unfortunately coyote uses IDLgr**** procedures that GDL does not provide.
